### PR TITLE
Feat/close buffer from picker

### PR DESCRIPTION
**Pull request by EricHenry**
Closes: #1593 

Hey all, this allows the user to delete a buffer from the file picker by using `ctrl-x` to close and `ctrl-X` to force close. I didn't update any documentation yet; I first wanted to get feedback on my approach and the keybindings that I used. 

Please let me know any ideas on how to improve this implementation.

### CODE REVIEW

File: helix-term/src/ui/picker.rs [hunk 1]
```diff
@@ -330,13 +338,15 @@ pub struct Picker<T: Item> {
     show_preview: bool,
 
     callback_fn: Box<dyn Fn(&mut Context, &T, Action)>,
+    key_event_callback_fn: KeyEventCallback<T>,
```
Comment (andreytkachenko) [inline helix-term/src/ui/picker.rs:341]: maybe make it optional? like `Option<KeyEventCallback<T>>`?

**Comment by lukepighetti** [comment]
What about `C-d` instead of `C-x`? That will match the close buffer key in normal mode

**Comment by EricHenry** [comment]
@lukepighetti I initially wanted to use `c-d`, but it is already being used by the picker to scroll the page down.

File: helix-term/src/ui/picker.rs (hunk 1 above)
Comment (EricHenry) [reply helix-term/src/ui/picker.rs]: Moved things around, the key_event_callback is not longer part of the constructor, instead there is a separate function to set the key event callback function.

**Comment by EricHenry** [comment]
I think I have landed on a good implementation, let me know your thoughts. It adds infrastructure for any specific usage of the picker to define a special handling of key events that do not conflict with the default key events for the picker.

**Comment by the-mikedavis** [comment]
My hangup with this change is that I think it's better solved by #5505. I don't think the code added here will be reused in the solution for that and I'm fairly certain that #5055 will make it very easy to implement this.

**Comment by EricHenry** [comment]
@the-mikedavis hey! Thanks for the feedback. I totally see where you are coming from. #5505 would be a great enhancement and allow for this exact flexibility. 

If no one is working on that issue. I’d love to pick it up since I’ve been in that code with this PR. Do you have any guidance or thoughts on the implementation. You mentioned #5055, what are your thoughts on using the dynamic picker to achieve #5505. Would you like to see this PR left open and then updated once the other change is made?

P.S. let me know if you’d like to take the conversation to that issue.

**Comment by the-mikedavis** [comment]
Ah, 5055 was a typo. I meant 5505 😅

I've been thinking about #5505 and tinkering with ideas for a while and it's very tricky to solve. I wrote out some ideas in [this branch](https://github.com/helix-editor/helix/compare/md-compositor-key-remapping-idea) (see [`478fc9a`](https://github.com/helix-editor/helix/commit/478fc9aa2c27f0d41819ef4feb3b42984dd99fe1) in particular). Feel free to have a go at it but I suspect that this is something that may end up being a very large change, and we will probably want to discuss it or solve it internally since really large changes are hard to collaborate on.

**Comment by gabydd** [comment]
I tried to implement the ``AnyPicker`` here https://github.com/gabydd/helix/tree/md-compositor-key-remapping-idea very interesting problem, definitely a difficult one to design well, this way(see the commit on my branch that uses an ``Enum`` and ``Any``) seems better then changing the way ``Items`` work though.

**Comment by pascalkuthe** [comment]
This branch has gone quite stale. We also intend to solve this differently (see above) so I am closing this out. Thanks for contributing

**Pull request closed without merging**